bench_test.go

```diff
@@ -0,0 +1,652 @@
+	{"POST", "/repos/:owner/:repo/statuses/:ref"},
+
+	// Search
+	{"GET", "/search/repositories"},
+	{"GET", "/search/code"},
+	{"GET", "/search/issues"},
+	{"GET", "/search/users"},
+	{"GET", "/legacy/issues/search/:owner/:repository/:state/:keyword"},
+	{"GET", "/legacy/repos/search/:keyword"},
+	{"GET", "/legacy/user/search/:keyword"},
+	{"GET", "/legacy/user/email/:email"},
+
+	// Users
+	{"GET", "/users/:user"},
+	{"GET", "/user"},
+	//{"PATCH", "/user"},
+	{"GET", "/users"},
+	{"GET", "/user/emails"},
+	{"POST", "/user/emails"},
+	{"DELETE", "/user/emails"},
+	{"GET", "/users/:user/followers"},
+	{"GET", "/user/followers"},
+	{"GET", "/users/:user/following"},
+	{"GET", "/user/following"},
+	{"GET", "/user/following/:user"},
+	{"GET", "/users/:user/following/:target_user"},
+	{"PUT", "/user/following/:user"},
+	{"DELETE", "/user/following/:user"},
+	{"GET", "/users/:user/keys"},
+	{"GET", "/user/keys"},
+	{"GET", "/user/keys/:id"},
+	{"POST", "/user/keys"},
+	//{"PATCH", "/user/keys/:id"},
+	{"DELETE", "/user/keys/:id"},
+}
+
+var gplusAPIRoutes = []route{
+	// People
+	{"GET", "/people/:userId"},
+	{"GET", "/people"},
+	{"GET", "/activities/:activityId/people/:collection"},
+	{"GET", "/people/:userId/people/:collection"},
+	{"GET", "/people/:userId/openIdConnect"},
+
+	// Activities
+	{"GET", "/people/:userId/activities/:collection"},
+	{"GET", "/activities/:activityId"},
+	{"GET", "/activities"},
+
+	// Comments
+	{"GET", "/activities/:activityId/comments"},
+	{"GET", "/comments/:commentId"},
+
+	// Moments
+	{"POST", "/people/:userId/moments/:collection"},
+	{"GET", "/people/:userId/moments/:collection"},
+	{"DELETE", "/moments/:id"},
+}
+
+func calcMem(name string, load func()) {
+	m := new(runtime.MemStats)
+
+	// before
+	runtime.GC()
+	runtime.GC()
+	runtime.GC()
+	runtime.GC()
+	runtime.ReadMemStats(m)
+	before := m.HeapAlloc
+
+	load()
+
+	// after
+	runtime.GC()
+	runtime.GC()
+	runtime.GC()
+	runtime.GC()
+	runtime.ReadMemStats(m)
+	after := m.HeapAlloc
+
+	println(""+name+":", after-before, "Bytes")
+}
+
+func goyaveHandler(res *Response, req *Request) {
+	res.String(http.StatusOK, "OK")
+}
+
+func loadGoyaveRouter(routes []route) *Router {
+	config.Load()
+	router := newRouter()
+
+	for _, r := range routes {
+		router.Route(r.method, r.path, goyaveHandler, nil)
+	}
+
+	return router
+}
+
+func benchRoutes(b *testing.B, router *Router, routes []route) {
+	w := httptest.NewRecorder()
+	r, _ := http.NewRequest("GET", "/", nil)
+	u := r.URL
+	rq := u.RawQuery
+
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		for _, route := range routes {
+			r.Method = route.method
+			r.RequestURI = route.path
+			u.Path = route.path
+			u.RawQuery = rq
+			router.requestHandler(w, r, goyaveHandler, nil)
+		}
+	}
+}
+
+func Benchmark_Static(b *testing.B) {
+	println("# Static routes len:", len(staticRoutes))
+	calcMem("Goyave", func() {
+		staticGoyave = loadGoyaveRouter(staticRoutes)
+	})
```

Comment:
Loading shouldn't be part measured in the benchmark. Move initialization in an `init()` function.